bundle_install_process.go

```diff
@@ -15,69 +18,120 @@ type Executable interface {
 	Execute(pexec.Execution) error
 }
 
+//go:generate faux --interface VersionResolver --output fakes/version_resolver.go
+type VersionResolver interface {
+	Lookup() (version string, err error)
+	CompareMajorMinor(left string, right string) (bool, error)
+}
+
+//go:generate faux --interface Calculator --output fakes/calculator.go
+type Calculator interface {
+	Sum(paths ...string) (string, error)
+}
+
 type BundleInstallProcess struct {
-	executable Executable
-	logger     LogEmitter
+	executable      Executable
+	logger          LogEmitter
+	versionResolver VersionResolver
+	calculator      Calculator
 }
 
-func NewBundleInstallProcess(executable Executable, logger LogEmitter) BundleInstallProcess {
+func NewBundleInstallProcess(executable Executable, logger LogEmitter, versionResolver VersionResolver, calculator Calculator) BundleInstallProcess {
 	return BundleInstallProcess{
-		executable: executable,
-		logger:     logger,
+		executable:      executable,
+		logger:          logger,
+		versionResolver: versionResolver,
+		calculator:      calculator,
 	}
 }
 
-func (ip BundleInstallProcess) Execute(workingDir, gemLayersDir string) error {
-	buffer := bytes.NewBuffer(nil)
-	args := []string{"config", "path", gemLayersDir}
-
-	ip.logger.Subprocess("Running 'bundle %s'", strings.Join(args, " "))
-	err := ip.executable.Execute(pexec.Execution{
-		Args:   args,
-		Stdout: buffer,
-		Stderr: buffer,
-	})
+func (ip BundleInstallProcess) ShouldRun(layer packit.Layer, workingDir string) (bool, string, string, error) {
```

Comment:
Can you add a comment that lays out what `ShouldRun` is doing, and what all the return values are?

Reply:
I didn't realize we were gonna start documenting all of the source code. Do you want me to backfill documentation for everything?

Reply:
I've added a comment for the new `ShouldRun` method. Let me know what I should do for everything else.

Reply:
This wasn't my intent. At first glance I was confused why a function called `ShouldRun` was returning more than just a `bool`. There's a lot going on here, I just thought it would be helpful for clarity for someone reading our code. I don't think you need to backfill all the docs as a part of this (unless you want to)

Reply:
What you've added looks great!